src/tools/advanced_regression_analysis.ts

```
@@ -371,63 +507,41 @@ function formatResultHeader(
+  const ssTot = y.reduce((sum, val) => sum + (val - yMean) * (val - yMean), 0);
+  const rSquared = ssTot === 0 ? (ssRes <= PERFECT_FIT_TOLERANCE ? 1 : 0) : 1 - ssRes / ssTot;
 
-  const rSquared = 1 - ssRes / ssTot;
-
-  // Calculate adjusted R-squared
-  const n = y.length;
   const p = numVariables;
-  const adjustedRSquared = 1 - ((1 - rSquared) * (n - 1)) / (n - p - 1);
+  const dfResidual = n - p - 1;
+  const adjustedRSquared =
+    dfResidual > 0 ? 1 - ((1 - rSquared) * (n - 1)) / dfResidual : rSquared;
```

ESLint

[error] Delete `⏎···`

(prettier/prettier)